code/modules/food_and_drinks/food/snacks_pastry.dm

```diff
@@ -28,7 +28,7 @@
 	if(last_check_time + 50 < world.time)
 		if(ishuman(M))
 			var/mob/living/carbon/human/H = M
-			if(H.mind && H.mind.assigned_role == "Security Officer" && !H.has_trait(TRAIT_AGEUSIA))
+			if(H.mind && H.mind.assigned_role == "Security Officer, Warden, Head of Security, Detective" && !H.has_trait(TRAIT_AGEUSIA))
```

Comment:
This is how I know you didn't test it.

Reply:
What, I tested it and ate a donut at 40 HP,  non-frosted 42 44 46 HP each bite. That means it healed the one area that was harmed 2 hp at a time, at once. Same happend when I ate a donut as a Sec officer at the same even HP level.

Reply:
this isn't how it works
H.mind.assigned_role == "Security Officer" || H.mind.assigned_role == "Warden" etc etc
or H.mind.assigned_role in list("Security Officer", "Warden", etc etc)
not like this.